helper/trace/debug_trace.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package trace
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime/trace"
+	"time"
+)
+
+func StartDebugTrace(dir string, filePrefix string) (file string, stop func() error, err error) {
+	if dir == "" {
+		dir = path.Join(os.TempDir(), "vault-traces")
+	}
+
+	d, err := os.Stat(dir)
+	if err != nil && !os.IsNotExist(err) {
+		return "", nil, fmt.Errorf("failed to stat trace directory %q: %s", dir, err)
+	}
+
+	if !os.IsNotExist(err) && !d.IsDir() {
+		return "", nil, fmt.Errorf("trace directory %q is not a directory", dir)
+	}
+
+	if os.IsNotExist(err) {
+		if err := os.Mkdir(dir, 0o755); err != nil {
+			return "", nil, fmt.Errorf("failed to create trace directory %q: %s", dir, err)
+		}
+	}
+
+	traceFile, err := filepath.Abs(fmt.Sprintf("%s/%s-%s.trace", dir, filePrefix, time.Now().Format(time.RFC3339)))
```

Comment:
can we switch this to using `filepath.Join` to create the path, rather than `Sprintf`?